great_expectations/core/expectation_validation_result.py

```diff
@@ -456,7 +464,13 @@ def __repr__(self):
     def __str__(self):
         return json.dumps(self.to_json_dict(), indent=2)
 
+    @public_api
```

Comment:
This is the only function I didn't annotate the return type on. There is some more implementation/typing work that needs to be done to get this correct because of he current implementation.